### ISSUE TITLE
Does the YAML stream save function keep indentation?

### ISSUE THREAD

**Issue by codec-abc** [aaubry/YamlDotNet#635]
Hi,

I want to do the following:
1. Deserialize 2 very similar YAML files without known properties (ie, the model isn't known to me).
2. Compare them and change properties (only existing scalar values) of the 2nd file based on the comparison with the first file.
3. Reserialize and save the 2nd file. But the goal here is to minimize the diff in the text output once serialized. 

I can do 1 and 2 using a ``YamlStream`` and implement a ``IYamlVisitor`` on the document, which shouldn't be too hard to implement. I can also reserialize the file using the ``Save`` method of the ``YamlStream``. Yet, I am wondering does deserializing and reserializing a YamlStream is guaranteed to keep the same indentation, order and so on?

**Comment by aaubry** [comment]
Hello,
The order and the style (plain, quoted, etc) should be preserved, but the indentation will not be preserved. That information is lost during parsing.

**Comment by codec-abc** [comment]
I somehow found a solution. I run my own "serializer" thanks to the IYamlVisitor. Basically I just got a copy of the original string. And when I visit a node I just count how many characters I should have written until this point. There I just copy the relevant characters from the source using the Start mark, And for ``YamlScalarNode`` I just use the value I fixed before. It seems to nice for what I try to achieve.
